### PR TITLE
Update BitbucketV2Proxy to use more v2 api.

### PR DESCRIPTION
**Pull request by shrimpy**

### CODE REVIEW

File: BitbucketV2Proxy.cs [hunk 1]
```diff
@@ -177,15 +193,15 @@ public string GetOAuthUri(string state = null, string redirectUri = null)
...
+            var requestUri = BitbucketProxyHelper.GetRequestUri(ApiBaseUrl, repoUrl, "refs", "branches");
             using (var client = CreateHttpClient(accessToken))
             {
-                using (var response = await client.GetAsync(requestUri))
-                {
-                    var info = await this.ProcessResponse<BitbucketProxy.BitbucketBranchesTagsInfo>("ListBranches", response);
-                    return info.branches;
-                }
+                List<BitbucketV2Branch> branchResults = await ListPagingItems<BitbucketV2Branch>(client, requestUri, "ListBranches");
+                branches.AddRange(branchResults.Select(v => v.ToBranchInfo()));
```
Comment (suwatch) [inline BitbucketV2Proxy.cs]: return branchResults.Select(v => v.ToBranchInfo()).ToArray();

File: BitbucketV2Proxy.cs [hunk 2]
```diff
@@ -324,43 +340,110 @@ public string GetOAuthUri(string state = null, string redirectUri = null)
...
+        public async Task<bool> RemoveService(string repoUrl, string accessToken, string serviceUrl)
+        {
+            CommonUtils.ValidateNullArgument("repoUrl", repoUrl);
+            CommonUtils.ValidateNullArgument("accessToken", accessToken);
+            CommonUtils.ValidateNullArgument("serviceUrl", serviceUrl);
+
+            var service = await GetService(repoUrl, accessToken, serviceUrl);
+            if (service != null)
+            {
+                var requestUri = BitbucketProxyHelper.GetRequestUri(APiV1BaseUrl, repoUrl, "services", service.id);
```
Comment (suwatch) [inline BitbucketV2Proxy.cs:384]: There is no V2 api to remove the old Service hook?

File: BitbucketV2Proxy.cs [hunk 3]
```diff
@@ -324,43 +340,110 @@ public string GetOAuthUri(string state = null, string redirectUri = null)
...
                 }
             }
+
+            return results;
         }
 
-        private async Task<BitbucketProxy.BitbucketHookInfo> GetWebHookInfo(string repoUrl, string accessToken, string hookUrl)
+        private async Task<BitbucketProxy.BitbucketSSHKeyFullInfo> GetSSHKey(string repoUrl, string accessToken, string sshKey)
+        {
+            var requestUri = BitbucketProxyHelper.GetRequestUri(APiV1BaseUrl, repoUrl, "deploy-keys");
```
Comment (suwatch) [inline BitbucketV2Proxy.cs:418]: no v2 api?

File: BitbucketV2Proxy.cs (hunk 2 above)
Comment (shrimpy) [inline BitbucketV2Proxy.cs:384]: nope. for services we have to go back to v1 API. 

---

In reply to: [64696631](https://github.com/suwatch/AzureSourceControls/pull/4#discussion_r64696631) [](ancestors = 64696631)

File: BitbucketV2Proxy.cs (hunk 3 above)
Comment (shrimpy) [inline BitbucketV2Proxy.cs:418]: not from documentation, i am waiting for Bitbucket to reply. Most likely i will create a bug on their repo to ask for this.

---

In reply to: [64696670](https://github.com/suwatch/AzureSourceControls/pull/4#discussion_r64696670) [](ancestors = 64696670)

Comment (shrimpy) [inline BitbucketV2Proxy.cs:418]: Created below bug

[https://bitbucket.org/site/master/issues/12746/missing-v2-api-to-add-remove-deployment](https://bitbucket.org/site/master/issues/12746/missing-v2-api-to-add-remove-deployment)

---

In reply to: [64789996](https://github.com/suwatch/AzureSourceControls/pull/4#discussion_r64789996) [](ancestors = 64789996,64696670)

**Pull request merged**